### PR TITLE
Add SO_REUSEADDR support, and ability to stop a server listening

### LINKED ISSUE #101: [Network] no way to set SO_REUSEADDR
example : 
```hack
$tcp = TCP\Server::createAsync($ipv4, '127.0.0.1', 8080);
// results in ADDRINUSE, how to stop the first one ?
// $tcp = TCP\Server::createAsync($ipv4, '127.0.0.1', 8080);
```

### PR DESCRIPTION
**Pull request by fredemmott**
fixes #101

### CODE REVIEW

**Comment by azjezz** [comment]
maybe we can add a property in the serve ( e.g `private bool $stopped = false` ), set it to `true` when `stopListening` is called, and throw an exception from `nextConnection` next time its called ?

File: src/network/Server.php [hunk 1]
```diff
@@ -44,4 +44,7 @@ public function nextConnectionNDAsync(): Awaitable<TNDSock>;
 
   /** Return the local (listening) address for the server */
   public function getLocalAddress(): this::TAddress;
+
+  /** Stop listening; open connection are not closed */
+  public function stopListening(): void;
```
Comment (azjezz) [inline src/network/Server.php]: just ... `stop(): void` ?

Comment (fredemmott) [reply src/network/Server.php]: I was worried that might imply that it stops active connections, which is not the case.

Comment (azjezz) [reply src/network/Server.php]: makes sense :+1:

**Pull request closed without merging**